### DISCUSSION TITLE
scph tutorial

### DISCUSSION THREAD

**Discussion by aebichan** [ttadano/alamode#80] [General]
Dear Dr. Tadano,

I am planning to use Alamode to evaluate the effects of anharmonicity in water.
In particular, I would like to perform SCPH calculation. 
I have tried doing it so far by intuition because I cannot seem to find a tutorial on how to perform SCPH calculation.
I would like to ask if you could kindly share your tutorial for scph calculation, or if it does not exist yet, a brief outline on how to perform it.
Another thing is that, if I understand correctly, only the loop diagrams (quartic anharmonicities) are considered in scph, is it also possible to include the bubble diagrams in the renormalized phonon dispersion? 

Thank you very much in advance!

Best regards,

Elvis

**Comment by ttadano** [comment]
For the tutorial on SCPH calculation, please have a look at the materials at https://github.com/ttadano/alamode_tutorial/tree/main/hands-on_TSUBAME2022.

> Another thing is that, if I understand correctly, only the loop diagrams (quartic anharmonicities) are considered in scph, is it also possible to include the bubble diagrams in the renormalized phonon dispersion?

Yes, it is possible. Please add `BUBBLE = 1` (or `BUBBLE = 2 or 3`) in the `&scph` field. This option uses the `EPSILON` tag, so you may need to adjust it appropriately (The default value is `EPSILON = 10` cm^-1) This feature is still experimental, so please use it with special care.

**Reply by aebichan** [reply]
Thank you very much for sparing your time in answering my question.
My main query is actually the preparation of anharmonic IFCs which in the tutorial was skipped understandably due to it being time-consuming. I had some ideas on how to proceed based on the papers listed in ppt and would like to confirm with you if they are correct.
1) Perform aimd and select a number of structures that are equally spaced in time.
2) Displace each of those structures by 0.1 Angstrom and calculate all configurations by DFT
3) Extract/collect all xmls (I'm using vasp) from all DFT calculations into a DFSET file. (<=== I am unsure of this one)
4) Then estimate the force constants using enet or adaptive lasso.

**Reply by ttadano** [reply]
Yes, that procedure is what I used for SrTiO3 in the tutorial. The way to generate the training snapshots is not unique, though. Instead of running AIMD, one can generate structure snapshots using the probability distribution function of harmonic oscillator given that the harmonic phonon is dynamically stable.

**Reply by aebichan** [reply]
Thank you very much for your prompt response!
